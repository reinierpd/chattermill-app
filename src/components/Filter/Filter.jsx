import React from 'react';
import PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import InputLabel from '@material-ui/core/InputLabel';
import FormControl from '@material-ui/core/FormControl';
import styled from 'styled-components';

const SelectContainer = styled.div`
  padding: 5px 0;
  margin-bottom: 10px;
`;

/**
 * @description
 * Component for create a select input filter.
 * @param {Object} - - Passed Props.
 * @returns {*} - React component.
 * */
const Filter = ({ data, name, label, value, handleChange }) => (
  <SelectContainer>
    <FormControl>
      <InputLabel htmlFor={`${name}-select`}>{label}</InputLabel>
      <Select
        native
        value={value || ''}
        onChange={ev => handleChange({ name, value: ev.target.value })}
        inputProps={{
          name: 'name',
          id: `${name}-select`,
        }}
      >
        <option value="" />
        {data.map(item =>
          typeof item === 'object' ? (
            <option key={item.id} value={item.id}>
              {item.name}
            </option>
          ) : (
            <option key={item} value={item}>
              {item}
            </option>
          ),
        )}
      </Select>
    </FormControl>
  </SelectContainer>
);

/**
 * @description
 * Filter component propTypes.
 * @param data - select options data. This can be an array of object or an
 * array of values.
 * @param name - input name.
 * @param value - input value
 * @param label - input label
 * @param handleChange - callback Function
 * */
Filter.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOf([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};
export default Filter;
