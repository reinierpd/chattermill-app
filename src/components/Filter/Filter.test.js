/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
// jest
import { mount } from 'enzyme';
import rendered from 'react-test-renderer';
import 'jest-styled-components';
import Filter from './Filter';

describe('AdDetail component', () => {
  it('validate correct mounting with data array of objects', () => {
    const data = [{ id: 1, name: 'Item1' }, { id: 2, name: 'Item2' }];
    const wrapper = rendered.create(
      <Filter
        data={data}
        name="item"
        label="Render item"
        value=""
        handleChange={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('validate correct mounting with data array of values', () => {
    const data = ['Item1', 'Item2'];
    const wrapper = rendered.create(
      <Filter
        data={data}
        name="item"
        label="Render item"
        value=""
        handleChange={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('validate callback called when select item', () => {
    const data = [{ id: 1, name: 'Item1' }, { id: 2, name: 'Item2' }];
    const mockCallback = jest.fn();
    const wrapper = mount(
      <Filter
        data={data}
        name="item"
        label="Render item"
        value=""
        handleChange={mockCallback}
      />,
    );
    wrapper
      .find('#item-select')
      .get(0)
      .props.onChange({ target: { name: 'item', value: '1' } });

    expect(mockCallback).toBeCalledWith({
      name: 'item',
      value: '1',
    });
  });
});
