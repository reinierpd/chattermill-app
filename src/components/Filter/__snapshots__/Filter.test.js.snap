// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdDetail component validate correct mounting 1`] = `
.c0 {
  padding: 5px 0;
  margin-bottom: 10px;
}

<div
  className="c0"
>
  <div
    className="MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="item-select"
    >
      Render item
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <select
        aria-invalid={false}
        className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        disabled={false}
        id="item-select"
        multiple={false}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        value=""
      >
        <option
          value=""
        />
        <option
          value={1}
        >
          Item1
        </option>
        <option
          value={2}
        >
          Item2
        </option>
      </select>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`AdDetail component validate correct mounting with data array of objects 1`] = `
.c0 {
  padding: 5px 0;
  margin-bottom: 10px;
}

<div
  className="c0"
>
  <div
    className="MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="item-select"
    >
      Render item
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <select
        aria-invalid={false}
        className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        disabled={false}
        id="item-select"
        multiple={false}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        value=""
      >
        <option
          value=""
        />
        <option
          value={1}
        >
          Item1
        </option>
        <option
          value={2}
        >
          Item2
        </option>
      </select>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`AdDetail component validate correct mounting with data array of values 1`] = `
.c0 {
  padding: 5px 0;
  margin-bottom: 10px;
}

<div
  className="c0"
>
  <div
    className="MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="item-select"
    >
      Render item
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <select
        aria-invalid={false}
        className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        disabled={false}
        id="item-select"
        multiple={false}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        value=""
      >
        <option
          value=""
        />
        <option
          value="Item1"
        >
          Item1
        </option>
        <option
          value="Item2"
        >
          Item2
        </option>
      </select>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
